### PR TITLE
Fix "no type named 'uint32_t'" failure in ICX build

### PR DESCRIPTION
**Pull request by Vika-F**
This PR replaces `std::uint32_t` and `std::int64_t` with `uint32_t` and `int64_t` in DAAL's 'algorithm_types.h' header file.

### CODE REVIEW

File: cpp/daal/include/algorithms/algorithm_types.h [hunk 1]
```diff
@@ -30,6 +30,8 @@
 #include "data_management/data/data_collection.h"
 #include "services/error_handling.h"
 #include "services/daal_shared_ptr.h"
+#include <stdint.h>
```
Comment (Alexandr-Solovev) [inline cpp/daal/include/algorithms/algorithm_types.h:33]: I doublechecked locally without `#include <stdint.h>` and it works also, should we delete this line?

Comment (Vika-F) [reply cpp/daal/include/algorithms/algorithm_types.h]: Yes, the lesser includes the better. I've removed this line.

**Review by Alexandr-Solovev — APPROVED** [review]

**Comment by Vika-F** [comment]
http://intel-ci.intel.com/ee749ce7-2537-f11e-81b8-a4bf010d0e2e

**Review by napetrov — APPROVED** [review]

**Pull request merged**